File: hackerrank/Algorithms/Implementation/Repeated String/solve.js
/**
 *      Author: Kessyus Fofano
 *      Email: kessyus@gmail.com
**/

'use strict';

const fs = require('fs');

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', function(inputStdin) {
    inputString += inputStdin;
});

process.stdin.on('end', function() {
    inputString = inputString.split('\n');

    main();
});

function readLine() {
    return inputString[currentLine++];
}

/*
 * Complete the 'repeatedString' function below.
 *
 * The function is expected to return a LONG_INTEGER.
 * The function accepts following parameters:
 *  1. STRING s
 *  2. LONG_INTEGER n
 */

function repeatedString(s, n) {
    // Write your code here
    if (n > s.length) {
        let num = 0;
        for (let i=0; i<s.length; i++)
            if (s[i]==='a')
                num++;
        let amount = parseInt(n / s.length);
        num *= amount;
        
        for (let i=0; i<(n % s.length); i++)
            if (s[i]==='a')
                num++;
        
        return num;    
    } else {
        let num = 0;
        for (let i=0; i<n; i++)
            if (s[i]==='a')
                num++;
        return num;
    }
}

function main() {
    const ws = fs.createWriteStream(process.env.OUTPUT_PATH);

    const s = readLine();

    const n = parseInt(readLine().trim(), 10);

    const result = repeatedString(s, n);

    ws.write(result + '\n');

    ws.end();
}

File: hackerrank/Algorithms/Implementation/Repeated String/solve.cpp
/**
 *      Author: Kessyus Fofano
 *      Email: kessyus@gmail.com
**/

#include <bits/stdc++.h>
using namespace std;

int main()
{
  ios_base::sync_with_stdio(false);
  cin.tie(0);

  string s;
  cin >> s;
  long long n;
  cin >> n;

  long long result = 0;

  if (n > s.length())
  {

    int countA = 0;
    // count 'a' in string
    for (auto i = 0; i < s.length(); i++)
    {
      if (s[i] == 'a')
        countA++;
    }

    // integer division
    long long letterA = (n / s.length()) * countA;

    // fraction
    long long fraction = (n % s.length());
    long long remainingA = 0;
    for (auto i = 0; i < fraction; i++)
    {
      if (s[i] == 'a')
        remainingA++;
    }

    result = letterA + remainingA;
  }
  else
  {

    int countA = 0;
    // count 'a' in string
    for (auto i = 0; i < n; i++)
    {
      if (s[i] == 'a')
        countA++;
    }

    result = countA;
  }

  cout << result << endl;

  return 0;
}
